BUILD

```diff
@@ -654,8 +654,8 @@ java_library(
     javacopts = select({
         "//:jdk9": ["--add-modules=jdk.unsupported"],
         "//conditions:default": [
-            "-source 7",
-            "-target 7",
+            "-source 8",
```

Comment:
Does this mean that we are going to drop Java 7 support?

Reply:
Yes.

Reply:
I don't think we would like to drop Java 7 support soon.

Reply:
The problem is: Java 11 doesn't support building with language level 7. Moreover, Bazel switched to using JDK 11 internally, and the build is spammed with:

```
  warning: -parameters is not supported for target value 1.7. Use 1.8 or later.
```

in many issue trackers: [1], [2].

[1] https://github.com/bazelbuild/bazel/issues/8772
[2] https://bugs.chromium.org/p/gerrit/issues/detail?id=11102

Reply:
I see. Since you are saying Java 11 doesn't support building language level 7 and Bazel switched to using JDK 11, does this mean there are currently no Bazel users who are using JDK 7?

Reply:
I apologize, my bad. Java 11, 13 and even 14 version still support java language level 7 (byte code major version 51).

This issue is entirely a Bazel bug (incompatibility of two Java compiler options) and should be be fixed in Bazel instead. Sorry for the noise.

Having said that, you should consider to discontinue Java 7 language level support in future Protobuf releases.

The most recent JDK 13 and upcoming JDK 14 release already issue this warning. JDK 13:

```
  $ cat A.java 
class A {}
  $ ~/pgm/jdk-13/bin/javac -source 7 -target 7 -parameters A.java
warning: [options] source value 7 is obsolete and will be removed in a future release
warning: [options] target value 7 is obsolete and will be removed in a future release
warning: -parameters is not supported for target value 7. Use 8 or later.
```

JDK 14:

```
  $ ~/pgm/jdk-14/bin/javac -source 7 -target 7 -parameters A.java
warning: [options] source value 7 is obsolete and will be removed in a future release
warning: [options] target value 7 is obsolete and will be removed in a future release
warning: -parameters is not supported for target value 7. Use 8 or later.
```